lib/frame.js

```diff
@@ -134,43 +165,42 @@ api.frame = (state, subjects, frame, parent, property = null) => {
       if(id in state.uniqueEmbeds[state.graph]) {
         _removeEmbed(state, id);
       }
-      state.uniqueEmbeds[state.graph][id] =
-        {parent, property};
     }
 
+    state.uniqueEmbeds[state.graph][id] = {parent, property};
+
     // push matching subject onto stack to enable circular embed checks
     state.subjectStack.push({subject, graph: state.graph});
 
     // subject is also the name of a graph
     if(id in state.graphMap) {
       let recurse = false;
       let subframe = null;
-      if(!('@graph' in frame)) {
+      if(!frame.hasOwnProperty('@graph')) {
         recurse = state.graph !== '@merged';
         subframe = {};
       } else {
         subframe = frame['@graph'][0];
+        recurse = !(id === '@merged' || id === '@default');
         if(!types.isObject(subframe)) {
           subframe = {};
         }
-        recurse = !(id === '@merged' || id === '@default');
       }
 
       if(recurse) {
         state.graphStack.push(state.graph);
-        state.graph = id;
         // recurse into graph
         api.frame(
-          state,
+          {...state, graph: id, embedded: false},
           Object.keys(state.graphMap[id]).sort(), [subframe], output, '@graph');
-        state.graph = state.graphStack.pop;
+        state.graphStack.pop;
```

Comment:
Should this be calling pop?  Should it have been calling it before?

Reply:
We pushed on to the stack before recursing, so need to pop it back off when done. We don't need to save that, since `state.graph` was set just for the call.

Reply:
Before it was setting `state.graph` to the pop function.  Now it's just doing nothing.  I assume this should be `pop()`, but it doesn't seem to matter either way in the test suite.

Reply:
No, you're right, should be `pop()` (Ruby-isms :) ). It doesn't matter for the tests, but it might in production.

I'll test before committing an update

Reply:
Fixing and filing an issue about it.